### PR TITLE
feat(build): Upgrade build process to emit ESM modules and a UMD bundle

### PR DESCRIPTION
**Pull request by brandonroberts**

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  dest: './release/bundles/router-store.umd.js',
+  format: 'umd',
+  moduleName: 'ngrx.router-store',
+  globals: {
+    'rxjs/Observable': 'Rx',
+    'rxjs/Subscriber': 'Rx',
+    'rxjs/operator/distinctUntilChanged': 'Rx.Observable.prototype',
+    'rxjs/operator/do': 'Rx.Observable.prototype',
+    'rxjs/operator/filter': 'Rx.Observable.prototype',
+    'rxjs/operator/map': 'Rx.Observable.prototype'
```
Comment (MikeRyanDev) [inline rollup.config.js]: You will need to map @ngrx/core here

``` ts
'@ngrx/core': 'ngrx.core'
```

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export default {
+  entry: './release/index.js',
+  dest: './release/bundles/router-store.umd.js',
+  format: 'umd',
+  moduleName: 'ngrx.router-store',
```
Comment (MikeRyanDev) [inline rollup.config.js]: The hyphen here won't work, you will need to camel-case this to `ngrx.routerStore`

File: tsconfig.legacy.json [hunk 3]
```diff
@@ -1,22 +1,23 @@
+
```
Comment (MikeRyanDev) [inline tsconfig.legacy.json]: This legacy file was only needed for @ngrx/core to preserve deep imports. You should not need a legacy build step.

**Review by MikeRyanDev — CHANGES_REQUESTED** [review]

File: rollup.config.js (hunk 2 above)
Comment (brandonroberts) [reply rollup.config.js]: Fixed

File: rollup.config.js (hunk 1 above)
Comment (brandonroberts) [reply rollup.config.js]: Fixed

File: tsconfig.legacy.json (hunk 3 above)
Comment (brandonroberts) [reply tsconfig.legacy.json]: Fixed

**Review by MikeRyanDev — APPROVED** [review]

**Pull request merged**